engine/minskyTensorOps.cc

```diff
@@ -433,6 +433,41 @@ namespace minsky
         hc.xvectors.insert(hc.xvectors.begin(), xv2.begin()+1, xv2.end());        
         hc.xvectors.insert(hc.xvectors.begin(), xv1.begin(), xv1.end()-1);
         cachedResult.hypercube(move(hc));
+        
+        set<size_t> newIdx;
+        vector<size_t> hcIdx1,hcIdx2;       
+        size_t tmpLinealIdx;        
+        
+        if (!arg1->index().empty() && !arg2->index().empty())
+        {
+        for (auto& i: arg1->index()) 
+            if (i!=arg1->index()[arg1->index().size()-1])
+            {
+			   for (auto& k: arg1->hypercube().splitIndex(i))
+			      hcIdx1.push_back(k);          
```

Comment:
This is suboptimal. Just use assignment, the compiler will probably use move assignment.